### ISSUE TITLE
SpaceInsideBlockBraces regression since 0.90.0

### ISSUE THREAD

**Issue by rdunlop** [rubocop-hq/rubocop#8799]
On 0.89.1, I would get multiple correction recommendations (and auto-corrections) for this cop.
On 0.90.0, I only get 1 correction recommendation each time I run `rubocop`.

I need to run rubocop multiple times in order to correct all of the offences.

The reproduction scenario I've been using is:
```ruby
# frozen_string_literal: true

describe User do
  each { puts }

  each {puts 3}
end
```

Expected:
```
spec/models/sample_spec.rb:6:9: C: Layout/SpaceInsideBlockBraces: Space missing inside {.
  each {puts 3}
        ^
spec/models/sample_spec.rb:6:15: C: Layout/SpaceInsideBlockBraces: Space missing inside }.
  each {puts 3}
              ^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable
```

Actual:
```
spec/models/sample_spec.rb:6:9: C: Layout/SpaceInsideBlockBraces: Space missing inside {.
  each {puts 3}
        ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

NOTE: it seems to be important that there be the success case before the error case.

```
$ [bundle exec] rubocop -V
0.92.0 (using Parser 2.7.1.5, rubocop-ast 0.6.0, running on ruby 2.5.3 x86_64-darwin17)
```
